patterns/direct_processor.py

```diff
@@ -0,0 +1,142 @@
+# request message carrying a string, the processor converts the payload to uppercase (for simplicity) and publishes to a 
+# designated topic.
+#   direct_requestor: 
+#       subscribes to request topic 'solace/samples/python/direct/processor/input/{count}
+#       proceses the payload
+#       publishes to reply topic 'solace/samples/python/direct/processor/output'
+
+TOPIC_PREFIX = "solace/samples/python"
+
+# Broker Config. Note: Could pass other properties Look into
+broker_props = {
+    "solace.messaging.transport.host": os.environ.get('SOLACE_HOST') or "tcp://localhost:55555,tcp://localhost:55554",
+    "solace.messaging.service.vpn-name": os.environ.get('SOLACE_VPN') or "default",
+    "solace.messaging.authentication.scheme.basic.username": os.environ.get('SOLACE_USERNAME') or "default",
+    "solace.messaging.authentication.scheme.basic.password": os.environ.get('SOLACE_PASSWORD') or ""
+    }
+
+# Build A messaging service with a reconnection strategy of 20 retries over an interval of 3 seconds
+# Note: The reconnections strategy could also be configured using the broker properties object
+messaging_service = MessagingService.builder().from_properties(broker_props)\
+                        .with_reconnection_retry_strategy(RetryStrategy.parametrized_retry(20,3))\
+                        .build()
+
+# Blocking connect thread
+messaging_service.connect()
+print(f'\nMessaging Service connected? {messaging_service.is_connected}')
+
+# Inner classes for connection error handling
+class ServiceEventHandler(ReconnectionListener, ReconnectionAttemptListener, ServiceInterruptionListener):
+    def on_reconnected(self, e: ServiceEvent):
+        print("\non_reconnected")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+    
+    def on_reconnecting(self, e: "ServiceEvent"):
+        print("\non_reconnecting")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+    def on_service_interrupted(self, e: "ServiceEvent"):
+        print("\non_service_interrupted")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+# Event Handling for the messaging service
+service_handler = ServiceEventHandler()
+messaging_service.add_reconnection_listener(service_handler)
+messaging_service.add_reconnection_attempt_listener(service_handler)
+messaging_service.add_service_interruption_listener(service_handler)
+ 
+# Inner classes for publish error handling
+class PublisherErrorHandling(PublishFailureListener):
+    def on_failed_publish(self, e: "FailedPublishEvent"):
+        print("on_failed_publish")
+
+# Handle received messages
+class MessageHandlerImpl(MessageHandler):
+    def on_message(self, message: InboundMessage):
+        out_topic = Topic.of(TOPIC_PREFIX + f'/direct/processor/output')
+        in_topic = message.get_destination_name()
+        payload_str = message.get_payload_as_string()
```

Comment:
What if the payload is not a string?

Reply:
You can do the type check trick here

```
payload = message.get_payload_as_string() if message.get_payload_as_string() != None else message.get_payload_as_bytes()
        if isinstance(payload, bytearray):
            print(f"Received a message of type: {type(payload)}. Decoding to string")
            payload = payload.decode()
print(f'Received input message (body string):\n{payload}')
```

Reply:
yeah. will make the change, but this could be done more elegantly either 
a) API handles necessary conversation and returns data in format as requested in the get_payload_xxx call or 
b) provide a convenience function to get the type 'xxx' and then let the user code do the necessary switch/case or if/else or even throw an error when unexpected type shows up!!

A topic for a different day... just making the change as you recommended for now.

Reply:
a) API is not really meant for payload processing, i.e. what comes in goes out. type conversation is considered payload processing. And if we would to suggest that to the API implementation then I assume it would require lots of edge case management...
b) that could be, however I doubt there is a function like that in the legacy APIs. The new gen APIs were really meant to achieve feature parity with the legacy APIs. This I believe would technically be a request to the legacy API

Reply:
New API can carry more useful convenience functions, which can be outside the legacy API parity aspect. API is all about making it easy/convenient for use for the developers - in this case, such a ternary (may be more if different or any type is possible) code to be repeated every time you access the payload. #justaying